Libraries/Utilities/utf8.js

```diff
@@ -0,0 +1,91 @@
+/**
+ * Copyright (c) 2016-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ *
+ * @providesModule utf8
+ * @flow
+ */
+'use strict';
+
+class ByteVector {
+  _storage: Uint8Array;
+  _sizeWritten: number;
+
+  constructor(size) {
+    this._storage = new Uint8Array(size);
+    this._sizeWritten = 0;
+  }
+
+  push(value: number): ByteVector {
+    const i = this._sizeWritten;
+    if (i === this._storage.length) {
+      this._realloc();
+    }
+    this._storage[i] = value;
+    this._sizeWritten = i + 1;
+    return this;
+  }
+
+  getBuffer(): ArrayBuffer {
+    return this._storage.buffer.slice(0, this._sizeWritten);
+  }
+
+  _realloc() {
+    const storage = this._storage;
+    this._storage = new Uint8Array(align(storage.length * 1.5));
+    this._storage.set(storage);
+  }
+}
+
+/*eslint-disable no-bitwise */
+exports.encode = (string: string): ArrayBuffer => {
```

Comment:
I might be missing something here, but doing utf8 ourself doesn't seem like something we should do.

Like the #1 rule of crypto "don't roll your own", I think this applies to string encoding too.